ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/KsqlRestClient.java

```diff
@@ -256,15 +284,22 @@ public void close() {
         throw closedIllegalStateException("close()");
       }
 
-      closed = true;
+      synchronized(this) {
+        closed = true;
+        this.notifyAll();
+      }
       responseScanner.close();
       response.close();
     }
 
-    private IllegalStateException closedIllegalStateException(String methodName) {
-      return new IllegalStateException("Cannot call " + methodName + " when QueryStream is closed");
+    private String closedIllegalStateMsg(String methodName) {
```

Comment:
Don't see why this method was extracted? Could just call the method below with the methodName

Reply:
java misunderstanding on my end. fixing.